cmd/cli/config/set.go

```diff
@@ -0,0 +1,160 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config"
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+	"github.com/bacalhau-project/bacalhau/pkg/logger"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+)
+
+func newSetCmd() *cobra.Command {
+	showCmd := &cobra.Command{
+		Use:   "set",
+		Args:  cobra.MinimumNArgs(2),
+		Short: "Set a value in the config.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return setConfig(args[0], args[1:]...)
+		},
+	}
+	return showCmd
+}
```

Comment:
Doesn't seem to work for update configuration.
```
% go run . config list | grep 'updateconfig.skipchecks'
 updateconfig.skipchecks                                         false                                    
% go run . config set updateconfig.skipchecks true     
Error: unsupported type <nil> for key: "updateconfig.skipchecks"
Usage:
  /var/folders/xz/qrd28pyx1095q2nzvhd9wb_c0000gn/T/go-build2679611526/b001/exe/bacalhau config set [flags]

Flags:
  -h, --help   help for set

Global Flags:
      --api-host string         The host for the client and server to communicate on (via REST).
                                Ignored if BACALHAU_API_HOST environment variable is set. (default "bootstrap.production.bacalhau.org")
      --api-port int            The port for the client and server to communicate on (via REST).
                                Ignored if BACALHAU_API_PORT environment variable is set. (default 1234)
      --log-mode logging-mode   Log format: 'default','station','json','combined','event' (default default)
      --repo string             path to bacalhau repo (default "/Users/simon/.bacalhau")

unsupported type <nil> for key: "updateconfig.skipchecks"
exit status 1
```

Reply:
`go run . config list | grep 'updateconfig.skipchecks'` doesn't return anything since that isn't a valid key.
I think what we mean is: `go run . config list | grep 'update.skipchecks'` which returns:
` update.skipchecks                                               false`

To continue with the example then:
```
go run . config set update.skipchecks true
frrist@oak ~/W/s/g/b/bacalhau (frrist/cmd/config)> cat ~/.bacalhau/config.yaml
update:
    skipchecks: true
```
This is good feedback though, we should return a better error in the event a user passes an invalid key.

Reply:
Hm, what I pasted above is literally the output from my terminal. Is it possible it is returning any YAML key (erroneously) contained in my config, even ones that aren't relevant?

Reply:
Haha yes it is 😂
```
% echo "OOGABOOGA: yes" >> ~/.bacalhau/config.yaml
% go run . config list | grep oogabooga
 oogabooga                                                       yes 
```
We should probably only be listing config keys that are recognised?

Reply:
ahh wow great catch, hand't considered that case. Thankfully is easy to address

Reply:
This has been fixed.